style/layer/highway_shield.js

```diff
@@ -1,49 +1,60 @@
 "use strict";
 
-export const interstate = {
-  id: "highway-shield-us-interstate",
-  type: "symbol",
-  paint: {
-    "text-color": "rgba(255, 255, 255, 1)",
-  },
-  filter: [
-    "all",
-    ["<=", "ref_length", 6],
-    ["==", "$type", "LineString"],
-    ["in", "network", "us-interstate"],
-  ],
-  layout: {
-    "icon-size": 2.0,
-    "text-font": ["Open Sans Bold"],
-    "text-size": 10.5,
-    "text-anchor": "center",
-    "text-field": "{ref}",
-    "text-offset": [0, 0],
-    "icon-image": "us_interstate",
-    "icon-text-fit": "both",
-    "icon-padding": 0,
-    "icon-text-fit-padding": [2.5, 0.6, 0, 0.5],
-    "symbol-spacing": {
-      stops: [
-        [7, 150],
-        [13, 300],
-        [16, 800],
-      ],
-    },
+import * as Util from "../js/util.js";
+
+function routeConcurrency(num) {
+  return [
+    "case",
+    ["!=", ["get", "route_" + num], null],
+    ["image", ["concat", "shield_", ["get", "route_" + num]]],
+    ["literal", ""],
+  ];
+}
+
+let shieldTextField = ["format"];
+for (var i = 1; i <= 5; i++) {
+  shieldTextField.push(routeConcurrency(i));
+  shieldTextField.push(" ");
```

Comment:
```suggestion
```

Reply:
We'll still need to pad out the shields to make proper spacing but committing this so we can move forward.

Reply:
FYI, adding padding to the shields would increase the likelihood of collisions with other icons and labels. If you find that you need to add padding only between shields but not completely around the shields, consider adding a transparent spacer GIF (technically a spacer PNG) between the shields inside the formatted text expression.

(I am simultaneously proud and ashamed of this suggestion.)